### PR TITLE
plugins: update memory addresses retrieval to use procptr32_t/procptr64_t.

### PR DESCRIPTION
**Pull request by davidebeatrici**
Update to almost all plugins to fit the new changes done to memory addresses retrieval system (https://github.com/mumble-voip/mumble/pull/2347).

Compile errors:

**AOC** at line 131:
> error: C2440: 'initializing' : cannot convert from 'BYTE *' to 'procptr32_t'
> There is no context in which this conversion is possible

**BF2** at lines 165 and 168:
> error: C2440: '=' : cannot convert from 'BYTE *' to 'procptr32_t'
> There is no context in which this conversion is possible

**BF4** at lines 47, 48 and 49 and **BF4_x86** at lines 45, 46 and 47:
> error: C2665: 'peekProc' : none of the 2 overloads could convert all the argument types
> ../mumble_plugin_win32.h(99): could be 'bool peekProc(procptr64_t,void *,SIZE_T)'
> ../mumble_plugin_win32.h(93): or       'bool peekProc(procptr32_t,void *,SIZE_T)'
> while trying to match the argument list '(procptr64_t, BYTE, int)'

**Borderlands 2** at lines 56 and 70:
> error: C2665: 'peekProc' : none of the 2 overloads could convert all the argument types
> ../mumble_plugin_win32.h(99): could be 'bool peekProc(procptr64_t,void *,SIZE_T)'
> ../mumble_plugin_win32.h(93): or       'bool peekProc(procptr32_t,void *,SIZE_T)'
> while trying to match the argument list '(procptr32_t, fetch::<unnamed-type-game_vects>, int)'

**CS** at line 138:
> error: C2440: '=' : cannot convert from 'BYTE *' to 'procptr32_t'
> There is no context in which this conversion is possible

**DYS** at line 131:
> error: C2440: 'initializing' : cannot convert from 'BYTE *' to 'procptr32_t'
> There is no context in which this conversion is possible

**GMod** at line 125:
> error: C2440: 'initializing' : cannot convert from 'BYTE *' to 'procptr32_t'
> There is no context in which this conversion is possible

**GTAIV** at lines 48 and 66:
> error: C2665: 'peekProc' : none of the 2 overloads could convert all the argument types
> ../mumble_plugin_win32.h(99): could be 'bool peekProc(procptr64_t,void *,SIZE_T)'
> ../mumble_plugin_win32.h(93): or       'bool peekProc(procptr32_t,void *,SIZE_T)'
> while trying to match the argument list '(procptr32_t, unsigned int, int)'

**GTAV** at line 22:
> error: C2665: 'peekProc' : none of the 2 overloads could convert all the argument types
> ../mumble_plugin_win32.h(99): could be 'bool peekProc(procptr64_t,void *,SIZE_T)'
> ../mumble_plugin_win32.h(93): or       'bool peekProc(procptr32_t,void *,SIZE_T)'

**Insurgency** at line 131:
> error: C2440: 'initializing' : cannot convert from 'BYTE *' to 'procptr32_t'
> There is no context in which this conversion is possible

**L4D2** at lines 151 and 158:
> error: C2440: 'initializing' : cannot convert from 'BYTE *' to 'procptr32_t'
> There is no context in which this conversion is possible

**QL** at line 32:
> error: C2665: 'peekProc' : none of the 2 overloads could convert all the argument types
> ../mumble_plugin_win32.h(99): could be 'bool peekProc(procptr64_t,void *,SIZE_T)'
> ../mumble_plugin_win32.h(93): or       'bool peekProc(procptr32_t,void *,SIZE_T)'
> while trying to match the argument list '(procptr32_t, BYTE, int)'

**UT3** at line 128:
> error: C2668: 'peekProc' : ambiguous call to overloaded function
> ../mumble_plugin_win32.h(129): could be 'T peekProc<procptr32_t>(procptr64_t)'
> with
> [
>     T=procptr32_t
> ]
> ../mumble_plugin_win32.h(122): or       'T peekProc<procptr32_t>(procptr32_t)'
> with
> [
>     T=procptr32_t
> ]
> while trying to match the argument list '(BYTE)'

**UT99** at line 102:
> error: C2665: 'peekProc' : none of the 2 overloads could convert all the argument types
> ../mumble_plugin_win32.h(99): could be 'bool peekProc(procptr64_t,void *,SIZE_T)'
> ../mumble_plugin_win32.h(93): or       'bool peekProc(procptr32_t,void *,SIZE_T)'
> while trying to match the argument list '(procptr32_t, char, int)'
At line 158:
> error: C2661: 'peekProc' : no overloaded function takes 2 arguments

### CODE REVIEW

File: plugins/bf1942/bf1942.cpp [hunk 1]
```diff
@@ -59,7 +58,7 @@ static int fetch(float *avatar_pos, float *avatar_front, float *avatar_top, floa
 
 	//peekProc(contextptr, ccontext, 128);
 
-	ok = peekProc((BYTE *) 0x00976274, avatar_pos, 12) &&
+	ok = peekProc((procptr32_t) 0x00976274, avatar_pos, 12) &&
```
Comment (mkrautz) [inline plugins/bf1942/bf1942.cpp]: Cast should not be necessary?

(Goes for a lot of places in throughout the PR)

**Comment by davidebeatrici** [comment]
Omitting it results in an error :wink:

**Comment by mkrautz** [comment]
Ah. The compiler can't choose the correct One. Hmm.

Maybe the solution is to have a separate header for 64-bit and 32-bit plugins...

**Comment by davidebeatrici** [comment]
Can't we use something like peekProc32 and peekProc64?

Both solutions are fine for me, but if we decide to use two headers we will have to update both when an edit needs to be made.

**Comment by davidebeatrici** [comment]
Compile error for **BF4**, **BF4_x86**, **Borderlands 2**, **GTAIV**, **GTAV**, **GW**, **L4D2**, **LOL**, **QL**, **STO** and **UT99**.:

> mumble_plugin_win32_ptr_type.h:83: error: C2664: 'BOOL ReadProcessMemory(HANDLE,LPCVOID,LPVOID,SIZE_T,SIZE_T *)': cannot convert argument 2 from 'procptr32_t' to 'LPCVOID'

**Comment by mkrautz** [comment]
@davidebeatrici Fixed in the PR. Please re-try.

**Comment by davidebeatrici** [comment]
Now they compile without any problem; I think that it's all ready to land :wink:

File: plugins/ut99/ut99.cpp [hunk 2]
```diff
@@ -114,9 +114,9 @@ static int fetch(float *avatar_pos, float *avatar_front, float *avatar_top, floa
 	float top_corrector[3];
 
 	// Peekproc and assign game addresses to our containers, so we can retrieve positional data
-	ok = peekProc((BYTE *) posptr, pos_corrector) &&
-	     peekProc((BYTE *) frtptr, front_corrector) &&
-	     peekProc((BYTE *) topptr, top_corrector);
+	ok = peekProc(posptr, pos_corrector, 12) &&
```
Comment (mkrautz) [inline plugins/ut99/ut99.cpp]: Size shouldn't be necessary.

File: plugins/ut99/ut99.cpp [hunk 3]
```diff
@@ -114,9 +114,9 @@ static int fetch(float *avatar_pos, float *avatar_front, float *avatar_top, floa
...
+	     peekProc(frtptr, front_corrector, 12) &&
```
Comment (mkrautz) [inline plugins/ut99/ut99.cpp]: Size shouldn't be necessary.

File: plugins/ut99/ut99.cpp [hunk 4]
```diff
@@ -114,9 +114,9 @@ static int fetch(float *avatar_pos, float *avatar_front, float *avatar_top, floa
...
+	     peekProc(topptr, top_corrector, 12);
```
Comment (mkrautz) [inline plugins/ut99/ut99.cpp]: Size shouldn't be necessary.

File: plugins/gw/gw.cpp [hunk 5]
```diff
@@ -137,15 +137,15 @@ static int fetch(float *avatar_pos, float *avatar_front, float *avatar_top, floa
 	char location;
 	int areaid;
 
-	ok = peekProc(camptr, cam) &&
-		 peekProc(posptr, pos) &&
-		 peekProc(camfrontptr, camfront) &&
+	ok = peekProc(camptr, cam, 12) &&
```
Comment (mkrautz) [inline plugins/gw/gw.cpp]: Size shouldn't be necessary.

File: plugins/gw/gw.cpp [hunk 6]
```diff
@@ -137,15 +137,15 @@ static int fetch(float *avatar_pos, float *avatar_front, float *avatar_top, floa
...
+		 peekProc(posptr, pos, 12) &&
```
Comment (mkrautz) [inline plugins/gw/gw.cpp]: Size shouldn't be necessary.

File: plugins/gw/gw.cpp [hunk 7]
```diff
@@ -137,15 +137,15 @@ static int fetch(float *avatar_pos, float *avatar_front, float *avatar_top, floa
...
+		 peekProc(camfrontptr, camfront, 12) &&
```
Comment (mkrautz) [inline plugins/gw/gw.cpp]: Size shouldn't be necessary.

**Comment by davidebeatrici** [comment]
I thought that I removed the size changes from every plugin... apparently I forgot some parts; done :wink:

**Comment by mkrautz** [comment]
Thanks! Landed via f28e9b7332aeb8710f6cf7020f28841096da365f.

**Pull request closed without merging**